--- a/kxsldbg/kxsldbgpart/CMakeLists.txt
+++ b/kxsldbg/kxsldbgpart/CMakeLists.txt
@@ -29,7 +29,7 @@
 kde4_add_plugin(kxsldbgpart WITH_PREFIX ${kxsldbgpart_PART_SRCS})
 
 
-target_link_libraries(kxsldbgpart ${KDE4_KTEXTEDITOR_LIBS} exslt ${LIBXML2_LIBRARIES} ${LIBXSLT_LIBRARIES} ${KDE4_KPARTS_LIBS} ${QT_QT3SUPPORT_LIBRARY})
+target_link_libraries(kxsldbgpart ${KDE4_KTEXTEDITOR_LIBS} exslt ${LIBXML2_LIBRARIES} ${LIBXSLT_LIBRARIES} ${KDE4_KPARTS_LIBS} ${QT_QT3SUPPORT_LIBRARY} ${CMAKE_THREAD_LIBS_INIT} )
 
 install(TARGETS kxsldbgpart  DESTINATION ${PLUGIN_INSTALL_DIR})
 
 
